ext/node/polyfills/worker_threads.ts

```diff
@@ -53,7 +54,16 @@ class _Worker extends EventEmitter {
       specifier = `data:text/javascript,${specifier}`;
     } else if (typeof specifier === "string") {
       specifier = resolve(specifier);
-      if (!specifier.toString().endsWith(".mjs")) {
+      let pkg;
+      try {
+        pkg = core.ops.op_require_read_closest_package_json(specifier);
+      } catch (_) {
+        // empty catch block when package json might not be present
+      }
+      if (
+        !(specifier.toString().endsWith(".mjs") ||
+          (pkg && pkg.exists && pkg.typ == "module"))
```

Comment:
I looks like brackets are missing here to extent the scope of the Negation in L64 over ( `.mjs`-suffix OR ESM-module), therefor `.js` files get wrapped by [broken] CJS support code, although they belong to an ESM module.